gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcReadChannel.java

```diff
@@ -55,6 +58,16 @@
   private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
   protected static final String METADATA_FIELDS = "contentEncoding,generation,size";
 
+  // ZeroCopy version of GetObjectMedia Method
+  private static final ZeroCopyMessageMarshaller getObjectMediaResponseMarshaller =
+      new ZeroCopyMessageMarshaller(GetObjectMediaResponse.getDefaultInstance());
+  private static final MethodDescriptor<GetObjectMediaRequest, GetObjectMediaResponse>
+      getObjectMediaMethod =
+          StorageGrpc.getGetObjectMediaMethod().toBuilder()
+              .setResponseMarshaller(getObjectMediaResponseMarshaller)
+              .build();
+  private static final boolean useZeroCopyMarshaller = ZeroCopyReadinessChecker.isReady();
```

Comment:
Why do we need to have this check?  It seems like we should say that next version of GCSIO depends on the gRPC 1.39.0+ and Protobuf 3.17.4+ and expect that ZeroCopy functionality is always available.

 Otherwise it will be hard to understand what's happening at runtime in addition to having a more complicated code to support.

Reply:
That's true for Dataproc but this isn't necessarily true for other cases that use gcsio such as Beam SDK. Also this is an experimental API so those can be changed in future so it's important to check whether those APIs are available with the given gRPC Java and Protobuf.